dmenu-rs: 5.5.2 -> 5.5.3

benjaminedwardwebb opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

This upgrade fixes a couple issues with the existing package.

See https://github.com/Shizcow/dmenu-rs/issues/53
and https://github.com/Shizcow/dmenu-rs/issues/62

See https://github.com/Shizcow/dmenu-rs/releases/tag/5.5.3 for the release notes on GitHub (which just says something like "bug fixes") and see https://github.com/Shizcow/dmenu-rs/compare/5.5.2...5.5.3 for the diff.

fixes https://github.com/NixOS/nixpkgs/issues/293164

## Verification

Verified by running

```bash
nix-build --attr dmenu-rs
```

and executing each resultant executable

```bash
./result/bin/dmenu
./result/bin/dmenu_path
./result/bin/dmenu_run
./result/bin/stest
```

Also executed

```bash
nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"
```

and tested the `dmenu` executable in the `results` directory, though, to my knowledge, no packages actually depend on `dmenu-rs`. It's a graphical application written for X.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

AndersonTorres commented:
It does not fit contributing guidelines.

Drafting...

dotlambda commented on pkgs/applications/misc/dmenu-rs/default.nix:
```diff
@@ -19,13 +19,13 @@
     repo = pname;
     rev = version;
-    sha256 = "sha256-6yO2S6j/BD6x/bsuTFKAKvARl1n94KRiPwpmswmUOPU=";
+    sha256 = "sha256-05Ia+GHeL8PzOwR7H+NEVhKJVMPhlIaQLwGfvwOAl0g=";
```
```suggestion
    hash = "sha256-05Ia+GHeL8PzOwR7H+NEVhKJVMPhlIaQLwGfvwOAl0g=";
```

benjaminedwardwebb replied:
What's the difference? Or did the `fetchFromGitHub` signature change recently? Committing either way ...

dotlambda commented on pkgs/applications/misc/dmenu-rs/default.nix:
```diff
@@ -19,13 +19,13 @@
 
   src = fetchFromGitHub {
     owner = "Shizcow";
     repo = pname;
```
```suggestion
    repo = "dmenu-rs";
```
#277994

benjaminedwardwebb replied:
Thanks for including the justification! Committing ...

dotlambda commented:
Please squash and mark ready for review when you're done.

AndersonTorres reviewed (commented):
Is this still broken on ARM Linux?

benjaminedwardwebb committed:
dmenu-rs: 5.5.2 -> 5.5.3

See https://github.com/Shizcow/dmenu-rs/releases/tag/5.5.3

This upgrade fixes a couple bugs with the existing package.

See https://github.com/Shizcow/dmenu-rs/issues/53
and https://github.com/Shizcow/dmenu-rs/issues/62

benjaminedwardwebb commented:
> Is this still broken on ARM Linux?

I believe so.

I tried building the package with `NIXPKGS_ALLOW_BROKEN=1 nix-build --attr dmenu-rs` on an `aarch64-linux` NixOS system running as virtual machine on VMWare on a M1 Mac, and the build failed with same errors I recall seeing when trying this in the past.

benjaminedwardwebb commented:
@AndersonTorres @dotlambda Is there anything remaining you think I should address? Thanks for taking the time to review. :)
